pkg/pmem-csi-driver/pmem-csi-driver.go

```diff
@@ -258,6 +278,62 @@ func (pmemd *pmemDriver) registerNodeController() error {
 	return nil
 }
 
+// startScheduler starts the scheduler extender if it is enabled. It
+// logs errors and cancels the context when it runs into a problem,
+// either during the startup phase (blocking) or later at runtime (in
+// a go routine).
+func (pmemd *pmemDriver) startScheduler(ctx context.Context, cancel func(), rs *registryserver.RegistryServer) error {
+	if pmemd.cfg.schedulerListen == "" {
+		return nil
+	}
+
+	resyncPeriod := 1 * time.Hour
+	factory := informers.NewSharedInformerFactory(pmemd.cfg.client, resyncPeriod)
```

Comment:
Please limit all this Kubernetes specific code to the newly introduced scheduler package. Apparently this `startScheduler` function should be placed inside `scheduler`. So I would expect just something like:

```
.....
 if pmemd.cfg.schedulerListen != "" {
   err := scheduler.Start(ctx, cancel, rs, pmemd.cfg.schedulerListen)
   ....
}
```

Reply:
I won't be that simple because this `scheduler.Start` would need more parameters than just `rs` and `pmemd.cfg.schedulerListen`. The API will also become more complex because of the need to manage Go routines.

As Dave wrote in https://dave.cheney.net/2020/02/23/the-zen-of-go, "Leave concurrency to the caller". That's what I am doing here.

Reply:
Another way to describe the current approach is "the scheduler package contains the business logic, the pmem-csi-driver package the setup code". Both Goroutine handling and creation of client and informer instances fall under "setup code".

Reply:
@pohly my point is about using the Kubernetes specifics into core(pmem-csi-driver) package.

> Another way to describe the current approach is "the scheduler package contains the business logic, the pmem-csi-driver package the setup code". Both Goroutine handling and creation of client and informer instances fall under "setup code".

I agree for some level. But I feel like `NewSchedule()` only needs _optional_ `client`(other than DriverName, RegistryServer) from which it could set up required informers itself. I called `client` optional because it only requires for testing this package. In reality, it should depend on incluster-config to setting-up the client.  So that we could avoid all client-go API calls from here.

Reply:
This may be true right now, but may not be true forever: if we add another component which needs informers, then we'll want to set them up once here and then hand them around. So conceptually this is still the right place to make these calls.